// @paypal/paypal-server-sdk/dist/esm/models/setupTokenRequestCard.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { lazy, object, optional, string, unknown } from '../schema';
import { addressSchema } from './address';
import { cardBrandSchema } from './cardBrand';
import { setupTokenCardExperienceContextSchema, } from './setupTokenCardExperienceContext';
export const setupTokenRequestCardSchema = object({
    name: ['name', optional(string())],
    number: ['number', optional(string())],
    expiry: ['expiry', optional(string())],
    securityCode: ['security_code', optional(string())],
    brand: ['brand', optional(cardBrandSchema)],
    billingAddress: ['billing_address', optional(lazy(() => addressSchema))],
    verificationMethod: ['verification_method', optional(string())],
    experienceContext: [
        'experience_context',
        optional(lazy(() => setupTokenCardExperienceContextSchema)),
    ],
    networkTransactionReference: [
        'network_transaction_reference',
        optional(unknown()),
    ],
});
