// @paypal/paypal-server-sdk/dist/cjs/models/addressDetails.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.addressDetailsSchema = void 0;
var schema_1 = require("../schema");
var name_1 = require("./name");
var phone_1 = require("./phone");
exports.addressDetailsSchema = (0, schema_1.object)({
    addressLine1: ['address_line_1', (0, schema_1.optional)((0, schema_1.string)())],
    addressLine2: ['address_line_2', (0, schema_1.optional)((0, schema_1.string)())],
    adminArea2: ['admin_area_2', (0, schema_1.optional)((0, schema_1.string)())],
    adminArea1: ['admin_area_1', (0, schema_1.optional)((0, schema_1.string)())],
    postalCode: ['postal_code', (0, schema_1.optional)((0, schema_1.string)())],
    countryCode: ['country_code', (0, schema_1.string)()],
    name: ['name', (0, schema_1.optional)((0, schema_1.lazy)(function () { return name_1.nameSchema; }))],
    id: ['id', (0, schema_1.optional)((0, schema_1.string)())],
    company: ['company', (0, schema_1.optional)((0, schema_1.string)())],
    phone: ['phone', (0, schema_1.optional)((0, schema_1.string)())],
    phoneNumber: ['phone_number', (0, schema_1.optional)((0, schema_1.lazy)(function () { return phone_1.phoneSchema; }))]
});
