// @paypal/paypal-server-sdk/dist/cjs/models/paymentTokenResponsePaymentSource.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.paymentTokenResponsePaymentSourceSchema = void 0;
var schema_1 = require("../schema");
var applePayPaymentToken_1 = require("./applePayPaymentToken");
var cardPaymentToken_1 = require("./cardPaymentToken");
var paypalPaymentToken_1 = require("./paypalPaymentToken");
var venmoPaymentToken_1 = require("./venmoPaymentToken");
exports.paymentTokenResponsePaymentSourceSchema = (0, schema_1.object)({
    card: ['card', (0, schema_1.optional)((0, schema_1.lazy)(function () { return cardPaymentToken_1.cardPaymentTokenSchema; }))],
    paypal: ['paypal', (0, schema_1.optional)((0, schema_1.lazy)(function () { return paypalPaymentToken_1.paypalPaymentTokenSchema; }))],
    venmo: ['venmo', (0, schema_1.optional)((0, schema_1.lazy)(function () { return venmoPaymentToken_1.venmoPaymentTokenSchema; }))],
    applePay: ['apple_pay', (0, schema_1.optional)((0, schema_1.lazy)(function () { return applePayPaymentToken_1.applePayPaymentTokenSchema; }))],
    bank: ['bank', (0, schema_1.optional)((0, schema_1.dict)((0, schema_1.unknown)()))]
});
