// @paypal/paypal-server-sdk/dist/cjs/models/paypalPaymentTokenUsagePattern.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.paypalPaymentTokenUsagePatternSchema = exports.PaypalPaymentTokenUsagePattern = void 0;
var schema_1 = require("../schema");
/**
 * Enum for PaypalPaymentTokenUsagePattern
 */
var PaypalPaymentTokenUsagePattern;
(function (PaypalPaymentTokenUsagePattern) {
    PaypalPaymentTokenUsagePattern["Immediate"] = "IMMEDIATE";
    PaypalPaymentTokenUsagePattern["Deferred"] = "DEFERRED";
    PaypalPaymentTokenUsagePattern["RecurringPrepaid"] = "RECURRING_PREPAID";
    PaypalPaymentTokenUsagePattern["RecurringPostpaid"] = "RECURRING_POSTPAID";
    PaypalPaymentTokenUsagePattern["ThresholdPrepaid"] = "THRESHOLD_PREPAID";
    PaypalPaymentTokenUsagePattern["ThresholdPostpaid"] = "THRESHOLD_POSTPAID";
})(PaypalPaymentTokenUsagePattern = exports.PaypalPaymentTokenUsagePattern || (exports.PaypalPaymentTokenUsagePattern = {}));
/**
 * Schema for PaypalPaymentTokenUsagePattern
 */
exports.paypalPaymentTokenUsagePatternSchema = (0, schema_1.stringEnum)(PaypalPaymentTokenUsagePattern, true);
