// @paypal/paypal-server-sdk/dist/cjs/models/vaultVenmoRequest.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
exports.__esModule = true;
exports.vaultVenmoRequestSchema = void 0;
var schema_1 = require("../schema");
var vaultedDigitalWalletShippingDetails_1 = require("./vaultedDigitalWalletShippingDetails");
var vaultVenmoExperienceContext_1 = require("./vaultVenmoExperienceContext");
exports.vaultVenmoRequestSchema = (0, schema_1.object)({
    description: ['description', (0, schema_1.optional)((0, schema_1.string)())],
    shipping: [
        'shipping',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return vaultedDigitalWalletShippingDetails_1.vaultedDigitalWalletShippingDetailsSchema; })),
    ],
    permitMultiplePaymentTokens: [
        'permit_multiple_payment_tokens',
        (0, schema_1.optional)((0, schema_1.boolean)()),
    ],
    usageType: ['usage_type', (0, schema_1.optional)((0, schema_1.string)())],
    customerType: ['customer_type', (0, schema_1.optional)((0, schema_1.string)())],
    experienceContext: [
        'experience_context',
        (0, schema_1.optional)((0, schema_1.lazy)(function () { return vaultVenmoExperienceContext_1.vaultVenmoExperienceContextSchema; })),
    ]
});
